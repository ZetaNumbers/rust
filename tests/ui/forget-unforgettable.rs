//@ build-fail

#![feature(unforgettable_types, forget_unsized)]

#[derive(Default)]
struct Unforgettable {
    _unforgettable: std::marker::PhantomUnforgettable,
}

fn main() {
    std::mem::forget(Unforgettable::default());
    std::mem::forget_unsized(Unforgettable::default());
}
