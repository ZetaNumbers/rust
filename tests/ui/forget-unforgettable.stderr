error[E0277]: `PhantomUnforgettable` cannot be forgotten
  --> $SRC_DIR/core/src/mem/manually_drop.rs:LL:COL
   |
   = note: `PhantomUnforgettable` cannot be forgotten
   |
   = help: within `Unforgettable`, the trait `Forget` is not implemented for `PhantomUnforgettable`, which is required by `Unforgettable: Forget`
note: required because it appears within the type `Unforgettable`
  --> $DIR/forget-unforgettable.rs:6:8
   |
LL | struct Unforgettable {
   |        ^^^^^^^^^^^^^

note: the above error was encountered while instantiating `fn std::mem::ManuallyDrop::<Unforgettable>::new`
  --> $SRC_DIR/core/src/mem/mod.rs:LL:COL

error[E0277]: `PhantomUnforgettable` cannot be forgotten
  --> $SRC_DIR/core/src/mem/mod.rs:LL:COL
   |
   = note: `PhantomUnforgettable` cannot be forgotten
   |
   = help: within `Unforgettable`, the trait `Forget` is not implemented for `PhantomUnforgettable`, which is required by `Unforgettable: Forget`
note: required because it appears within the type `Unforgettable`
  --> $DIR/forget-unforgettable.rs:6:8
   |
LL | struct Unforgettable {
   |        ^^^^^^^^^^^^^

note: the above error was encountered while instantiating `fn std::mem::forget_unsized::<Unforgettable>`
  --> $DIR/forget-unforgettable.rs:12:5
   |
LL |     std::mem::forget_unsized(Unforgettable::default());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
